paasta_tools/cli/cmds/mark_for_deployment.py

```diff
@@ -1323,143 +1323,100 @@ def check_if_instance_is_done(
-            "{}.{} in {} is marked as stopped. Marked as deployed.".format(
-                service, status.instance, cluster
-            )
-        )
+
+    # Case: instance is stopped
+    if status.expected_instance_count == 0 or status.desired_state == "stop":
+        log.debug(f"{inst_str} is marked as stopped. Ignoring it.")
         return True
-    else:
-        if long_running_status.app_count != 1:
-            print(
-                "  {}.{} on {} is still bouncing, {} versions "
-                "running".format(
-                    service, status.instance, cluster, long_running_status.app_count,
-                )
-            )
-            return False
-        if not git_sha.startswith(status.git_sha):
-            print(
-                "  {}.{} on {} doesn't have the right sha yet: {}".format(
-                    service, instance, cluster, status.git_sha,
-                )
-            )
-            return False
-        if long_running_status.deploy_status not in [
-            "Running",
-            "Deploying",
-            "Waiting",
-        ]:
+
+    short_git_sha = git_sha[:8]
+    active_shas = {g[:8] for g, c in status.active_shas}
+    if short_git_sha in active_shas:
+        non_desired_shas = active_shas.difference({short_git_sha})
+        # Case: bounce in-progress
+        if len(non_desired_shas) == 1:
+            (other_sha,) = non_desired_shas
             print(
-                "  {}.{} on {} isn't running yet: {}".format(
-                    service, instance, cluster, long_running_status.deploy_status,
-                )
+                f"  {inst_str} is still bouncing, from {other_sha} to {short_git_sha}"
             )
             return False
 
-        # The bounce margin factor defines what proportion of instances we need to be "safe",
-        # so consider it scaled up "enough" if we have that proportion of instances ready.
-        required_instance_count = int(
-            math.ceil(
-                instance_config.get_bounce_margin_factor()
-                * long_running_status.expected_instance_count
-            )
-        )
-        if required_instance_count > long_running_status.running_instance_count:
+        # Case: previous bounces not yet finished when this one was triggered
+        elif len(non_desired_shas) > 1:
             print(
-                "  {}.{} on {} isn't scaled up yet, "
-                "has {} out of {} required instances (out of a total of {})".format(
-                    service,
-                    instance,
-                    cluster,
-                    long_running_status.running_instance_count,
-                    required_instance_count,
-                    long_running_status.expected_instance_count,
-                )
+                f"  {inst_str} hasn't started bouncing to {short_git_sha}, because a previous "
+                f"bounces aren't done yet, with these other versions still running: {non_desired_shas}"
```

Comment:
This implies that we have to wait for previous bounces to finish before starting to deploy the new version, which I'm pretty sure isn't true. Once we see `short_git_sha` in `active_shas`, we've started launching the new version.